lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4452,6 +4457,60 @@ Instruction *SPIRVToLLVM::transOCLBuiltinFromExtInst(SPIRVExtInst *BC,
   return CI;
 }
 
+void SPIRVToLLVM::transAuxDataInst(SPIRVExtInst *BC) {
+  assert(BC->getExtSetKind() == SPIRV::SPIRVEIS_Aux_Data);
+  if (!BC->getModule()->preserveAllFunctionAttributesAndMetadata())
+    return;
+  auto Args = BC->getArguments();
+  // Args 0 and 1 are common between attributes and metadata.
+  // 0 is the function, 1 is the name of the attribute/metadata as a string
+  auto *SpvFcn = BC->getModule()->getValue(Args[0]);
+  auto *F = static_cast<Function *>(getTranslatedValue(SpvFcn));
+  assert(F && "Function should already have been translated!");
```

Comment:
Should we really assert that or may be return void is better?

Reply:
imo we should assert because i have some code that skips processing the inst if the metadata/attr already exists, if the fcn wasnt translated that logic doesn't work